packages/react-theme-provider/src/makeClasses.ts

```diff
@@ -0,0 +1,88 @@
+import { IStyle } from '@uifabric/merge-styles';
+import { ITheme } from '@fluentui/theme';
+import { makeStyles } from './makeStyles';
+
+/**
+ * The `makeClasses` helper encapsulates `makeStyles`, and given a style map which follows
+ * a specific naming convention, produces a hook function which takes in the component
+ * draft state and auto distributes classes into the draft state. This removes a lot of
+ * boilerplate code using `classnames` helpers to manage distributing classnames manually.
+ *
+ * Usage:
+ *
+ * ```tsx
+ * const useButtonClasses makeClasses(theme => {
+ *   root: { ... },
+ *   _primary: { ... },
+ *   _size_small: { ... }
+ * });
+ * ```
+ *
+ * The class naming convention is broken down as follows:
+ *
+ * * No underscores - a slot class name. (E.g. "root", "icon", etc)
+ * * Prefixed with underscore - a modifier. (E.g. "_primary", "_fluid")
+ * * Contains 2 underscores - a name/value matcher. (E.g. "_size_small")
+ *
+ * Modifier classnames are added to the root className when the state contains a truthy value
+ * of the same name. For example, when the primary flag is present, the "_primary" modifier
+ * class will be appended to the root className prop.
+ *
+ * Enum classnames are also added to the root className when teh state contains an enum value
+ * which matches the value in the className. for example, when the `size` enum value is `small`,
+ * the "_size_small" enum class will be appended to the root className prop.
+ */
+export const makeClasses = <TStyleSet extends { [key: string]: IStyle }>(
```

Comment:
i think this belongs better in `react-compose`

Reply:
hmm I'm not convinced that's the right place. 

It depends on makeStyles, which depends on theme.

It relies on the renderer, provided by the themeprovider, to make the styles. (Separate PR.)

My thought is that makeClasses which is in react-compose should be removed. Let react-compose be about utilities that help with composition.